backend/pkg/engine/processor.go

```diff
@@ -43,14 +43,18 @@ func (p *Processor) Run(
 	resources []unstructured.Unstructured,
 	oldResources []unstructured.Unstructured,
 ) (*Results, error) {
+	response := &Results{
+		PolicyValidations: validatePolicies(policies),
+	}
+	if len(response.PolicyValidations) > 0 {
+		return response, nil
+	}
```

Comment:
Maybe returning an custom Error with the violations as property instead of adding it to the results. So it can be directly handled as error.